Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -125,3 +208,299 @@ theorem IsTranscendenceBasis.isAlgebraic_field {F E : Type*} {x : ι → E}
+
+theorem matroid_base_iff {s : Set A} :
+    (matroid inj).Base s ↔ IsTranscendenceBasis R ((↑) : s → A) := Iff.rfl
+
+theorem matroid_cRank_eq : (matroid inj).cRank = trdeg R A :=
+  (trdeg_eq_iSup_cardinalMk_isTranscendenceBasis _).symm
+
+end
+
+theorem matroid_basis_iff [IsDomain A] {s t : Set A} : (matroid inj).Basis s t ↔
+    s.AlgebraicIndependent R ∧ s ⊆ t ∧ ∀ a ∈ t, IsAlgebraic (adjoin R s) a := by
+  rw [Matroid.Basis, maximal_iff_forall_insert fun s t h hst ↦ ⟨h.1.subset hst, hst.trans h.2⟩]
+  simp_rw [matroid_indep_iff, ← and_assoc, matroid_e, subset_univ, and_true]
+  exact and_congr_right fun h ↦ ⟨fun max a ha ↦ of_not_not fun tr ↦ max _
+    (fun ha ↦ tr (isAlgebraic_algebraMap (⟨a, subset_adjoin ha⟩ : adjoin R s)))
+      ⟨.insert h.1 tr, insert_subset ha h.2⟩,
+    fun alg a ha h ↦ ((insert_iff ha).mp h.1).2 (alg _ <| h.2 <| mem_insert ..)⟩
+
+theorem matroid_basis_iff_of_subsingleton [Subsingleton A] {s t : Set A} :
+    (matroid inj).Basis s t ↔ s = t := by
+  have := inj.subsingleton
+  simp_rw [Matroid.Basis, matroid_indep_iff, AlgebraicIndependent.of_subsingleton, true_and,
+    matroid_e, subset_univ, and_true, ← le_iff_subset, maximal_le_iff]
+
+theorem isAlgebraic_adjoin_iff_of_matroid_basis [NoZeroDivisors A] {s t : Set A} {a : A}
+    (h : (matroid inj).Basis s t) : IsAlgebraic (adjoin R s) a ↔ IsAlgebraic (adjoin R t) a := by
+  cases subsingleton_or_nontrivial A
+  · apply iff_of_false <;> apply is_transcendental_of_subsingleton
+  have := (isDomain_iff_noZeroDivisors_and_nontrivial A).mpr ⟨inferInstance, inferInstance⟩
+  exact ⟨(·.adjoin_of_forall_isAlgebraic fun x hx ↦ (hx.2 <| h.1.1.2 hx.1).elim),
+    (·.adjoin_of_forall_isAlgebraic fun x hx ↦ ((matroid_basis_iff inj).mp h).2.2 _ hx.1)⟩
+
+theorem matroid_closure_eq [IsDomain A] {s : Set A} :
+    (matroid inj).closure s = algebraicClosure (adjoin R s) A := by
+  have ⟨B, hB⟩ := (matroid inj).exists_basis s
+  simp_rw [← hB.closure_eq_closure, hB.1.1.1.closure_eq_setOf_basis_insert, Set.ext_iff, mem_setOf,
+    matroid_basis_iff, ← matroid_indep_iff inj, hB.1.1.1, subset_insert, true_and, SetLike.mem_coe,
+    mem_algebraicClosure, ← isAlgebraic_adjoin_iff_of_matroid_basis inj hB, forall_mem_insert]
+  exact fun _ ↦ and_iff_left fun x hx ↦ isAlgebraic_algebraMap (⟨x, subset_adjoin hx⟩ : adjoin R B)
+
+theorem matroid_flat_iff [IsDomain A] {s : Set A} :
+    (matroid inj).Flat s ↔ ∃ S : Subalgebra R A, S = s ∧ ∀ a : A, IsAlgebraic S a → a ∈ s := by
+  rw [Matroid.flat_iff_closure_eq, matroid_closure_eq]
+  set S := algebraicClosure (adjoin R s) A
+  refine ⟨fun eq ↦ ⟨S.restrictScalars R, eq, fun a (h : IsAlgebraic S _) ↦ ?_⟩, ?_⟩
+  · rw [← eq]; exact h.restrictScalars (adjoin R s)
+  rintro ⟨s, rfl, hs⟩
+  refine Set.ext fun a ↦ ⟨(hs _ <| adjoin_eq s ▸ ·), fun h ↦ ?_⟩
+  exact isAlgebraic_algebraMap (A := A) (by exact (⟨a, subset_adjoin h⟩ : adjoin R s))
+
+theorem matroid_flat_of_subsingleton [Subsingleton A] (s : Set A) : (matroid inj).Flat s := by
```

Comment:
This, and the next lemma, is saying that your matroid is `Simple` - i.e.  pairs are independent  https://github.com/apnelson1/Matroid/blob/main/Matroid/Simple.lean